### ISSUE TITLE
[5pt] src optimization error HydroID for calib not found

### ISSUE THREAD

**Issue by RobHanna-NOAA** [NOAA-OWP/inundation-mapping#1215]
This one may not necessary need be a bug, but needs to be evaluated to see if a bigger issue exists. During a full BED run of 4.5.2.5, hundreds, true count unknown, recorded the following errors:

ERROR: HydroID for calb point was not found in the hydrotable (check hydrotable)

ie) 
/outputs/fim_4_5_2_5/logs/src_optimization/log_spatial_src_adjust.log:12633:ERROR: HydroID for calb point was not found in the hydrotable (check hydrotable) for HUC: 04300108  branch id: 2502000002 hydroid: 15190069
/outputs/fim_4_5_2_5/logs/src_optimization/log_spatial_src_adjust.log:12634:ERROR: HydroID for calb point was not found in the hydrotable (check hydrotable) for HUC: 04300108  branch id: 2502000002 hydroid: 15190069
/outputs/fim_4_5_2_5/logs/src_optimization/log_spatial_src_adjust.log:12635:ERROR: HydroID for calb point was not found in the hydrotable (check hydrotable) for HUC: 04300108  branch id: 2502000002 hydroid: 15190070
/ou

I didn't scan that file very much but it was clear there were other hucs such as 01090004.  I can find more applicable hucs and suspect it could be a significant number.  

This was discovered by the new post_processing error scan for all of the log files including all logs being created in post processing. However, this error has not been seen before this build.  There are no other log details available but the code that throughs it can be found at `src_roughness_optimizations.py`

It might be that this needs to be changed to use word warning instead of error, but the fact that it just started suggests a bigger issue.

**Comment by mluck** [comment]
This might be a result of dropping DEM-derived reaches that fail to crosswalk.

**Comment by CarsonPruitt-NOAA** [comment]
I was thinking the same thing. If that's the case, this error may be okay. We'll just have to account for the HydroID drop in the future so that this doesn't error out.

**Comment by mluck** [comment]
> This might be a result of dropping DEM-derived reaches that fail to crosswalk.

#1205 for reference

**Comment by RobHanna-NOAA** [comment]
After more analysis of the logs, I don't think this bug is related to the [1212 ](https://github.com/NOAA-OWP/inundation-mapping/issues/1212) Crosswalk bug. HUCs showing errors in 1212 are generally not on the list of HUCs related to this issue. 

But. I found out which HUCs have this problem. They are:
01090004
03030007  (only had a few)
04300108  (lots and lots)
07120006  (only had a few)
12040104  (only had a few)
13080001  (the error above, PLUS... a new bug?)

13080001 does appear to have a related issue with 1212. I recorded its details in the 1212 card.

For 04300108, I think it a complete breakdown and possible all of its branches failed which is why he picked up the error above and more.

**Comment by mluck** [comment]
This is happening because the branches use `nwm_subset_streams_levelPaths`, and since the DEM-derived reaches are dropped if they don't snap within 100 meters of the NWM streams, the hydroIDs are missing where the levelpaths are clipped or dropped in waterbodies. So when `src_roughness_optimization.py` checks if the calibration point is in the branch hydroTable (https://github.com/NOAA-OWP/inundation-mapping/blob/dev/src/src_roughness_optimization.py#L203), it doesn't exist and the error is logged.

There is no other handling of the error and the loop continues iterating, so the "ERROR" message could be changed to "WARNING" as suggested in the original issue. Another option might be to filter the calibration points dataframe so that it only contains HydroIDs found in the hydrotable.

**Comment by RobHanna-NOAA** [comment]
Yup. That sounds good to me.  :)  To me: ERROR means "fix me". Warning mean, be aware of it, but it is just fine.  Thanks Matt.

**Comment by RobHanna-NOAA** [comment]
hummmm.... picked some new errors against 04300108.  Do I need to bring one of the other WIP PR before I test this.
Let me know if you need more log detail on this.

/outputs/Rob_test_src_opt/logs/unit/04300108_unit.log:1520:fiona._err.CPLE_OpenFailedError: /fim_temp/Rob_test_src_opt/04300108/branches/2502000025/gw_catchments_reaches_filtered_addedAttributes_crosswalked_2502000025.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/unit/04300108_unit.log:1541:fiona.errors.DriverError: /fim_temp/Rob_test_src_opt/04300108/branches/2502000025/gw_catchments_reaches_filtered_addedAttributes_crosswalked_2502000025.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/unit/04300108_unit.log:1567:***** An error has occured  *****
/outputs/Rob_test_src_opt/logs/unit/04300108_unit.log:2398:fiona._err.CPLE_OpenFailedError: /fim_temp/Rob_test_src_opt/04300108/branches/4591000003/gw_catchments_reaches_filtered_addedAttributes_crosswalked_4591000003.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/unit/04300108_unit.log:2419:fiona.errors.DriverError: /fim_temp/Rob_test_src_opt/04300108/branches/4591000003/gw_catchments_reaches_filtered_addedAttributes_crosswalked_4591000003.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/unit/04300108_unit.log:2445:***** An error has occured  *****
/outputs/Rob_test_src_opt/logs/branch/04300108_branch_2502000025.log:258:fiona._err.CPLE_OpenFailedError: /fim_temp/Rob_test_src_opt/04300108/branches/2502000025/gw_catchments_reaches_filtered_addedAttributes_crosswalked_2502000025.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/branch/04300108_branch_2502000025.log:279:fiona.errors.DriverError: /fim_temp/Rob_test_src_opt/04300108/branches/2502000025/gw_catchments_reaches_filtered_addedAttributes_crosswalked_2502000025.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/branch/04300108_branch_4591000003.log:254:fiona._err.CPLE_OpenFailedError: /fim_temp/Rob_test_src_opt/04300108/branches/4591000003/gw_catchments_reaches_filtered_addedAttributes_crosswalked_4591000003.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/branch/04300108_branch_4591000003.log:275:fiona.errors.DriverError: /fim_temp/Rob_test_src_opt/04300108/branches/4591000003/gw_catchments_reaches_filtered_addedAttributes_crosswalked_4591000003.gpkg: No such file or directory
/outputs/Rob_test_src_opt/logs/log_bankfull_indentify.log:42:ERROR --> 04300108  branch id: 2502000027 failed (details: *** cannot set a frame with no defined index and a scalar  File "/foss_fim/src/identify_src_bankfull.py", line 216, in src_bankfull_lookup

--------
the 0400108_branch_2502000025.log file has this (notice.. burn bridges??)
-----------------------------------------------------------------
Burn in bridges 04300108 2502000025
/usr/local/lib/python3.10/dist-packages/rasterstats/io.py:335: NodataWarning: Setting nodata to -999; specify nodata explicitly
  warnings.warn(
Traceback (most recent call last):
  File "fiona/_shim.pyx", line 83, in fiona._shim.gdal_open_vector
  File "fiona/_err.pyx", line 291, in fiona._err.exc_wrap_pointer
fiona._err.CPLE_OpenFailedError: /fim_temp/Rob_test_src_opt/04300108/branches/2502000025/gw_catchments_reaches_filtered_addedAttributes_crosswalked_2502000025.gpkg: No such file or directory

**Comment by mluck** [comment]
Those are all addressed by #1216 except 
```
/outputs/Rob_test_src_opt/logs/log_bankfull_indentify.log:42:ERROR --> 04300108 branch id: 2502000027 failed (details: *** cannot set a frame with no defined index and a scalar File "/foss_fim/src/identify_src_bankfull.py", line 216, in src_bankfull_lookup
```

**Comment by RobHanna-NOAA** [comment]
oh right.  Ok. I will load 1216 and check it, then merge it into this one and test again.. we shoul be good then. Thanks Matt

**Comment by mluck** [comment]
That last error is now fixed by #1216.
